pkg/storage/cacher_test.go

```diff
@@ -335,7 +338,6 @@ func TestFiltering(t *testing.T) {
 	}
 	defer watcher.Stop()
 
-	verifyWatchEvent(t, watcher, watch.Added, podFoo)
```

Comment:
since we're testing boundary conditions, we have to ensure the server has more data than it should return, and ensure the subset we expect is returned. otherwise, the server could be completely ignoring the passed resourceVersion and the test would still pass.